daemon/restart.go

```diff
@@ -49,7 +49,7 @@ func (daemon *Daemon) containerRestart(ctx context.Context, daemonCfg *configSto
 	// (implying also on Windows) as the HCS must have exclusive
 	// access to mount the containers filesystem inside the utility
 	// VM.
-	if !containertypes.Isolation.IsHyperV(actualIsolation) {
+	if !daemon.UsesSnapshotter() && !containertypes.Isolation.IsHyperV(actualIsolation) {
```

Comment:
We should probably update the comment above to describe why we're not doing it with containerd snapshotters as well now.

Reply:
Am I following this logic correctly and `daemon.Mount` is incrementing a reference count if the given container's storage is already mounted to the host, i.e. the current Linux case. The only way I understand this logic is if that's the case, and `daemon.containerStop` does an implicit `daemon.Unmount`, and `daemon.containerStart` does an implicit `daemon.Mount`.

The current change proposed here, as far as I understand, will reintroduce the needless (but otherwise harmless?) unmount/mount for Linux containerd-storage modes, as they still currently rely on host-mounting to function.

Would it be better to simply say "Only take a reference if the container is already mounted", or is that not knowable/thread-safe here? That would isolate this logic from host-mount-related flow-changes made elsewhere, e.g. moving to `WithSnapshot`.

Reply:
Actually, it looks like it could be caused by our ref-counted mount having a whitelist of filesystems. I opened a PR to reverse the logic and use the refcounted mounter by default (which we already talked about in the past): https://github.com/moby/moby/pull/47105

If these tests pass on the mentioned PR, we can close this one.